smdebug/pytorch/hook.py

```diff
@@ -118,6 +129,47 @@ def _prepare_collections(self):
 
     # This hook is invoked by trainer prior to running the forward pass.
     def forward_pre_hook(self, module, inputs):
+        if self.profiler_enabled and self.forward_start != 0:
+            self.timeline_writer.write_trace_events(
+               op_name="forward",
+               #timestamp=time.time(),
```

Comment:
I saw the negative timestamp too. will check how to fix it in the timeline writer

Reply:
fixed in the latest commit in timeline writer. you can now pass timestamp here

Reply:
Thanks for fixing it! I added the timestamp.